python/foglamp/services/core/api/support.py

```diff
@@ -89,6 +91,42 @@
     return web.json_response({"bundle created": bundle_name})
 
 
+async def get_syslog_entries(request):
+    """ Returns a list of syslog trail entries sorted with most recent first and total count
+        (including the criteria search if applied)
+
+    :Example:
+        curl -X GET http://localhost:8081/foglamp/syslog
+        curl -X GET "http://localhost:8081/foglamp/syslog?limit=5"
+        curl -X GET "http://localhost:8081/foglamp/syslog?source=Storage"
+        curl -X GET "http://localhost:8081/foglamp/syslog?limit=5&source=Storage"
+    """
+
+    try:
+        limit = int(request.query['limit']) if 'limit' in request.query and request.query['limit'] != '' else __DEFAULT_LIMIT
+        if limit < 0:
+            raise ValueError
+    except (Exception, ValueError):
+        raise web.HTTPBadRequest(reason="Limit must be a positive integer")
+
+    try:
+        source = request.query['source'] if 'source' in request.query and request.query['source'] != '' else 'FogLAMP'
+        if source not in ['FogLAMP', 'Storage', 'FogLAMP Storage']:
+            raise ValueError
+    except ValueError:
+        raise web.HTTPBadRequest(reason="{} is not a valid source".format(source))
+
+    try:
+        cmd = "grep -n '{}\[' {} | tail -n {}".format(source, _SYSLOG_FILE, limit)
+        a = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE).stdout.readlines()
+        c = []
+        for b in a:
+            c.append(b.decode())
```

Comment:
so above 3 lines ` [c = b.decode() for b in a]` ? please add a comment what `b.decode does here;

Reply:
done